File: src/main/java/org/akkreditierung/ui/model/AdvertiserConfigModelProvider.java
package org.akkreditierung.ui.model;

import org.apache.wicket.extensions.markup.html.repeater.data.sort.SortOrder;

public class AdvertiserConfigModelProvider extends
		GenericProvider<AdvertiserConfig> {

	private static final AdvertiserConfigBean ADVERTISER_CONFIG_BEAN = new AdvertiserConfigBean();

	private static final long serialVersionUID = -6117562733583734933L;

	public AdvertiserConfigModelProvider() {
		super(ADVERTISER_CONFIG_BEAN);
		setSort("id", SortOrder.ASCENDING);
	}
}

File: src/main/scala/org/akkreditierung/ui/model/AdvertiserConfigModelProvider.scala
package org.akkreditierung.ui.model

import com.avaje.ebean.Query
import org.apache.wicket.extensions.markup.html.repeater.data.sort.SortOrder

@SerialVersionUID(-6117562733583734933L)
class AdvertiserConfigModelProvider(filterContainer: DBFilter[AdvertiserConfig]) extends GenericProvider[AdvertiserConfig](new AdvertiserConfigBean) {
  setSort("fach", SortOrder.ASCENDING)

  override def filter(query: Query[AdvertiserConfig]): Query[AdvertiserConfig] = {
    if (filterContainer != null) filterContainer.apply(query)
    super.filter(query)
  }
}